runtime/lua/vim/shared.lua

```diff
@@ -4,34 +4,36 @@
 -- test-suite. If, in the future, Nvim itself is used to run the test-suite
 -- instead of "vanilla Lua", these functions could move to src/nvim/lua/vim.lua
 
+local shared = {}
 
 --- Returns a deep copy of the given object. Non-table objects are copied as
 --- in a typical Lua assignment, whereas table objects are copied recursively.
 ---
 --@param orig Table to copy
 --@returns New table of copied keys and (nested) values.
-local function deepcopy(orig)
-  error(orig)
-end
```

Comment:
I think we still need a "dummy" definition. When I ran `gen_vimdoc.py` from this PR, `deepcopy` was missing from `runtime/doc/if_lua.txt`.

```lua
function shared.deepcopy(orig) end  -- luacheck: no unused
```